src/index.ts

```diff
@@ -142,20 +148,8 @@ function anymatch(matchers: Matcher[], testString: string | undefined): boolean
   return matchPatterns(patterns, testString);
 }
 
-const flatten = (list: any[], result = []) => {
```

Comment:
`Array#flat` is supported since Node v12

https://node.green/#ES2019-features-Array-prototype--flat--flatMap-

Reply:
In the places where you call `.flat()`, I think you need to pass `.flat(Infinity)` to replicate the old behaviour. Though with that said, the types doesn't allow nested arrays in the first place so I'm not sure if flattening is needed, but maybe better to keep for compat?

Reply:
well `any[]` can as well be `any[][]` so it doesn't differ

and `flatten` is an internal util that was only used once so I don't think setting infinite depth makes sense here

Reply:
I don't mean `any[]`, but where it was being used and called here:

<img width="556" alt="image" src="https://github.com/user-attachments/assets/b3cc8c6d-e20b-47d9-ad68-112da97374c0">

Arguably flattening with any level doesn't make sense given the types, but if we ignore the types and assume people could nest with any depth, then `.flat(Infinity)` matches the previous behaviour.